lib/github.com/mpllang/mpllib/Hashset.sml

```diff
@@ -66,13 +65,8 @@ datatype 'a t =
           val current = Array.sub (data, i)
         in
           case current of
-            SOME y => if eq (x, y) then false else loop (i+1) (probes+1)
-          | NONE =>
-              if bcas (data, i) (current, SOME x) then
-                (* (Concurrency.faa sz 1; true) *)
-                true
-              else
-                loop i probes
+            SOME y =>  not (eq (x, y)) andalso loop (i+1) (probes+1)
+          | NONE => bcas (data, i) (current, SOME x) (* (Concurrency.faa sz 1; true) *) orelse loop i probes
```

Comment:
This was easier to read before the edit IMO.

Also, the commented-out code is left-over from a previous experiment to track sizes, which was useful, and I'd like to leave it in. This is clear before the edit, but afterwards looks strange.